app/routes/logout.tsx

```diff
@@ -1,18 +1,84 @@
-import { redirect } from '@remix-run/node';
-import { getUsername, logout } from '~/session.server';
+import { json, redirect } from '@remix-run/node';
+import { Form, Link, useLoaderData } from '@remix-run/react';
+import {
+  AbsoluteCenter,
+  Box,
+  Button,
+  ButtonGroup,
+  Center,
+  Container,
+  Stack,
+  Text,
+} from '@chakra-ui/react';
+import { createLogoutRequest } from '~/lib/saml.server';
+import { getUsername, sloUsernameCookie, logout } from '~/session.server';
 
-import type { ActionArgs } from '@remix-run/node';
+import type { ActionArgs, LoaderArgs } from '@remix-run/node';
 
 export const action = async ({ request }: ActionArgs) => {
   const user = await getUsername(request);
+  const formData = await request.formData();
 
-  // Invalidate the Starchart session but do not log out from Seneca IDP.
-  if (user) {
-    return logout(request);
+  const sloUsername = formData.get('sloUsername');
+  if (user !== undefined) {
```

Comment:
Is `undefined` important here (vs `null`)?  If you just mean "not set," it's rare to do this and I'd use `if (!user) {...}`.